### PR TITLE
[EXTREMELY WIP!!!] Meta+Toolchain: Bring jakt in as a part of the toolchain

### PR DESCRIPTION
**Pull request by alimpfard**
(Still WIP)

### CODE REVIEW

**Comment by alimpfard** [comment]
calm down stalebot, sheesh

**Comment by alimpfard** [comment]
Current status:
- [X] Cross-compiling toolchain
- [ ] CMake integration
	- [X] basic integration (niceties to compile)
	- [ ] dependency stuff
		- [X] generate a depfile
		- [ ] make sure it actually werks
	- [ ] convenience stuff: integration with existing serenity_app functionality
- [ ] Demos
	- [X] A very basic demo app
	- [ ] A reasonably complex demo app

**Comment by alimpfard** [comment]
Bad bot! No!

**Comment by alimpfard** [comment]
shoo, it's progressing at its own pace (of 1 feature/month)

**Comment by ADKaster** [comment]
Bad bot

File: Toolchain/BuildJakt.sh [hunk 1]
```diff
@@ -0,0 +1,193 @@
...
+                buildstep "system_headers" mkdir -p "$(dirname "Root/usr/include/$target")"
+                buildstep "system_headers" "$INSTALL" "$header" "Root/usr/include/$target"
+            done
+            unset SRC_ROOT
+        popd
+
+        rm -rf jakt
+        mkdir -p jakt
+
+        pushd jakt
```
Comment (ADKaster) [inline Toolchain/BuildJakt.sh:165]: surely we only need to build jakt that knows how to target $ARCH once, and then build_for should only apply to the shupport libs?  Or do we encode the sysroot in the jakt binary?

Comment (ADKaster) [reply Toolchain/BuildJakt.sh:165]: Or... are there compiler-specific dependencies in actually executing the jakt binary itself? surely not. only the support libs should be compiler-specific.... right?

**Comment by ADKaster** [comment]
re: reasonable demo app, why not use the ones we tried for the rust experiment?

- lscpu/lspci/lsusb
- Clipboard

**Comment by alimpfard** [comment]
Because they would be rewrites, and I have a strict no-rewrite personal policy for this PR

File: Meta/CMake/common_options.cmake [hunk 2]
```diff
@@ -26,7 +26,7 @@ serenity_option(ENABLE_ACCELERATED_GRAPHICS ON CACHE BOOL "Enable use of acceler
 
 serenity_option(HACKSTUDIO_BUILD OFF CACHE BOOL "Automatically enabled when building from HackStudio")
 
-serenity_option(ENABLE_JAKT OFF CACHE BOOL "Enable building jakt files")
+serenity_option(ENABLE_JAKT ON CACHE BOOL "Enable building jakt files")
```
Comment (alimpfard) [inline Meta/CMake/common_options.cmake:29]: I think this currently does all of nothing lol

File: Toolchain/BuildJakt.sh (hunk 1 above)
Comment (alimpfard) [reply Toolchain/BuildJakt.sh]: (yes, only the support libs are compiler-dependent)

File: Meta/CMake/utils.cmake [hunk 3]
```diff
@@ -88,10 +88,39 @@ function(serenity_libc target_name fs_name)
...
+            else()
+                list(APPEND cpp_sources ${source})
+            endif()
+        endforeach()
+
+        if (jakt_source)
+            if (ENABLE_JAKT)
+                message(STATUS "Skipping ${name} because jakt is disabled")
+                serenity_jakt_executable(${name} MAIN_SOURCE ${jakt_source} CPP_SOURCES ${cpp_sources})
+            endif()
```
Comment (ADKaster) [inline Meta/CMake/utils.cmake:113]: :buggieinspect:

File: Meta/CMake/common_options.cmake (hunk 2 above)
Comment (ADKaster) [reply Meta/CMake/common_options.cmake:29]: reset to default off? until we have something fun to show.

**Pull request closed without merging**